python/ray/serve/policy.py

```diff
@@ -1,188 +1,167 @@
+from abc import ABCMeta, abstractmethod
 from enum import Enum
 import itertools
 
 import numpy as np
 
-import ray
-from ray.serve.router import Router
 from ray.serve.utils import logger
 
 
-class RandomPolicyQueue(Router):
-    """
-    A wrapper class for Random policy.This backend selection policy is
-    `Stateless` meaning the current decisions of selecting backend are
-    not dependent on previous decisions. Random policy (randomly) samples
-    backends based on backend weights for every query. This policy uses the
-    weights assigned to backends.
+class RoutingPolicy:
+    """Defines the interface for a routing policy for a single endpoint.
+
+    To add a new routing policy, a class should be defined that provides this
+    interface. The class may be stateful, in which case it may also want to
+    provide a non-default constructor.
```

Comment:
Stateful but not the state will not be restored on router reconstruct.

Reply:
Good point